rolling-shutter/keyperimpl/shutterservice/syncmonitor.go

```
@@ -29,36 +32,87 @@ func (s *SyncMonitor) Start(ctx context.Context, runner service.Runner) error {
-
-			if currentBlockNumber > lastBlockNumber {
-				lastBlockNumber = currentBlockNumber
-			} else {
-				log.Error().
-					Int64("last-block-number", lastBlockNumber).
-					Int64("current-block-number", currentBlockNumber).
-					Msg("block number has not increased between checks")
-				return errors.New("block number has not increased between checks")
+				log.Debug().Err(err).Msg("skipping sync check due to error")
 			}
 		case <-ctx.Done():
 			log.Info().Msg("stopping syncMonitor due to context cancellation")
 			return ctx.Err()
 		}
 	}
 }
+
+var ErrBlockNotIncreasing = errors.New("block number has not increased between checks")
+
+func (s *SyncMonitor) runCheck(
+	ctx context.Context,
+	db *database.Queries,
+	keyperdb *keyperDB.Queries,
+	lastBlockNumber *int64,
+) error {
+	if s.dkgIsRunning(ctx, keyperdb) {
+		log.Debug().Msg("dkg is running, skipping sync monitor checks")
+		return nil
+	}
+
+	record, err := db.GetIdentityRegisteredEventsSyncedUntil(ctx)
+	if err != nil {
+		if errors.Is(err, pgx.ErrNoRows) {
+			log.Warn().Err(err).Msg("no rows found in table identity_registered_events_synced_until")
+			return nil // This is not an error condition that should stop monitoring
+		}
+		return fmt.Errorf("error getting identity_registered_events_synced_until: %w", err)
+	}
+
+	currentBlockNumber := record.BlockNumber
+	log.Debug().Int64("current-block-number", currentBlockNumber).Msg("current block number")
+
+	if currentBlockNumber > *lastBlockNumber {
+		*lastBlockNumber = currentBlockNumber
+		return nil
+	}
+
+	log.Error().
+		Int64("last-block-number", *lastBlockNumber).
+		Int64("current-block-number", currentBlockNumber).
+		Msg("block number has not increased between checks")
+	return ErrBlockNotIncreasing
+}
+
+func (s *SyncMonitor) dkgIsRunning(ctx context.Context, keyperdb *keyperDB.Queries) bool {
+	batchConfig, err := keyperdb.GetLatestBatchConfig(ctx)
```

In both DB queries here, we are silently returning an error and skipping the synch monitor, shouldn't we rather propagate the error instead of returning true? (similarly to what we're doing here: https://github.com/shutter-network/rolling-shutter/blob/main/rolling-shutter/keyper/keyper.go#L267)